tests/bluetooth/bsim_bt/bsim_test_mesh/CMakeLists.txt

```diff
@@ -29,10 +31,12 @@ else()
   src/test_friendship.c
   src/test_provision.c
   src/test_beacon.c
-	src/test_scanner.c
+  src/test_scanner.c
  )
 endif()
 
+zephyr_compile_options(-Werror)
```

Comment:
@trond-snekvik It is a bit too much to do this here. As one may want (say while debugging) to do something sloppy that causes warnings, and not have the compiler prevent you. In twister, it is twister the one raising warnings to errors, to ensure no warnings get pass.
About your question: `compile.sh` was done dropping the output unless there is errors to be silent (the "silence is golden" Unix rule), to not pollute your console, and to not pollute the CI output. It would be possible to add an option for compile.sh that sets `-Werror`. Say either just adding `cmake_args="-DEXTRA_CFLAGS=-Werror"` to that particular app line in compile.sh, or adding a new option overall for compile.sh that adds that same flag. (Note that compile.sh was born just as a script for CI )

Reply:
The main mode of work is the normal test running. I do not see any reason to hide warnings (that might be real bugs. Good example, where I made a bug and had to fix it emergently: https://github.com/zephyrproject-rtos/zephyr/pull/37079)
This is quite unfriendly to the new users. They do not know hidden assumptions and merge warnings and bugs in the main repo. It happens again and again. Seems Unix rule doesn't work well here.

Reply:
I'll revert my changes related to this. Submited bug for BubbleSim tests build system.
https://github.com/zephyrproject-rtos/zephyr/issues/38303